salt/modules/libcloud_storage.py

```diff
@@ -0,0 +1,352 @@
+# -*- coding: utf-8 -*-
+'''
+Connection module for Apache Libcloud Storage (object/blob) management for a full list
+of supported clouds, see http://libcloud.readthedocs.io/en/latest/storage/supported_providers.html
+
+Clouds include Amazon S3, Google Storage, Aliyun, Azure Blobs, Ceph, OpenStack swift
+
+.. versionadded:: Nitrogen
```

Comment:
I'm not sure this can still make it into `Nitrogen`, but probably into `Oxygen` -- but the maintainers will be able to confirm.